// jade-brunch/lib/index.js
// Generated by CoffeeScript 1.4.0
(function() {
  var JadeCompiler, jade, sysPath;

  jade = require('jade');

  sysPath = require('path');

  module.exports = JadeCompiler = (function() {

    JadeCompiler.prototype.brunchPlugin = true;

    JadeCompiler.prototype.type = 'template';

    JadeCompiler.prototype.extension = 'jade';

    function JadeCompiler(config) {
      this.config = config;
      return;
    }

    JadeCompiler.prototype.compile = function(data, path, callback) {
      var content, error, result, _ref, _ref1;
      try {
        content = jade.compile(data, {
          compileDebug: false,
          client: true,
          filename: path,
          pretty: !!((_ref = this.config.plugins) != null ? (_ref1 = _ref.jade) != null ? _ref1.pretty : void 0 : void 0)
        });
        return result = "module.exports = " + content + ";";
      } catch (err) {
        return error = err;
      } finally {
        callback(error, result);
      }
    };

    JadeCompiler.prototype.include = [sysPath.join(__dirname, '..', 'vendor', 'runtime.js')];

    return JadeCompiler;

  })();

}).call(this);
